developer/src/kmc-package/src/compiler/package-validation.ts

```diff
@@ -0,0 +1,55 @@
+import { KmpJsonFile, CompilerCallbacks } from '@keymanapp/common-types';
+import { CompilerMessages } from './messages.js';
+
+export class PackageValidation {
+
+  constructor(private callbacks: CompilerCallbacks) {
+  }
+
+  public validate(kmpJson: KmpJsonFile.KmpJsonFile) {
+    if(!this.checkKeyboards(kmpJson)) {
+      return false;
+    }
+    if(!this.checkLexicalModels(kmpJson)) {
+      return false;
+    }
+    return true;
+  }
+
+  private checkForDuplicatedLanguages(resourceType: 'keyboard'|'model', id: string, languages: KmpJsonFile.KmpJsonFileLanguage[]) {
+    let tags: {[index:string]: boolean} = {};
+    for(let lang of languages) {
+      const langTag = lang.id.toLowerCase();
+      if(tags[langTag]) {
+        this.callbacks.reportMessage(CompilerMessages.Warn_PackageShouldNotRepeatLanguages({resourceType:resourceType, id:id, tag:lang.id}));
+      } else {
+        tags[langTag] = true;
+      }
+    }
+  }
+
+  private checkLexicalModels(kmpJson: KmpJsonFile.KmpJsonFile): boolean {
+    if(kmpJson.lexicalModels?.length > 0 && kmpJson.keyboards?.length > 0) {
+      this.callbacks.reportMessage(CompilerMessages.Error_PackageCannotContainBothModelsAndKeyboards());
+      return false;
+    }
```

Comment:
Could this be split into a separate `checkForBothModelsAndKeyboards()` and then explicitly called in `validate`?